aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -794,12 +796,32 @@ internal class RealAWSCognitoAuthPlugin(
                     }
                     is AuthorizationState.Error -> {
                         token?.let(authStateMachine::cancel)
-                        onError.accept(
-                            CognitoAuthExceptionConverter.lookup(
-                                authZState.exception,
-                                "Fetch auth session failed."
+                        when (val error = authZState.exception) {
+                            is SessionError -> {
+                                val idpError = when (error.exception) {
+                                    is aws.sdk.kotlin.services.cognitoidentityprovider.model.NotAuthorizedException ->
+                                        SessionExpiredException()
+                                    is aws.sdk.kotlin.services.cognitoidentity.model.NotAuthorizedException ->
+                                        SignedOutException(SignedOutException.RECOVERY_SUGGESTION_GUEST_ACCESS_DISABLED)
+                                    is aws.sdk.kotlin.services.cognitoidentity.model.CognitoIdentityException ->
+                                        SignedOutException(
+                                            SignedOutException.RECOVERY_SUGGESTION_GUEST_ACCESS_POSSIBLE,
+                                            cause = CognitoAuthExceptionConverter.lookup(
+                                                error,
+                                                "Fetch auth session failed."
+                                            )
+                                        )
+                                    // TODO: session expired
```

Comment:
Is this TODO just for the hub event? If yes then why dont we add that in this PR?